locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -1071,15 +1195,32 @@ public boolean apply(@Nullable Method input) {
                                   // can coerce to
                                   if (input == null) return false;
                                   if (!input.getName().equals(option.getKey())) return false;
+                                  int numOptionParams = option.getValue() instanceof List ? ((List)option.getValue()).size() : 1;
                                   Type[] parameterTypes = input.getGenericParameterTypes();
-                                  return parameterTypes.length == 1
-                                          && TypeCoercions.tryCoerce(option.getValue(), TypeToken.of(parameterTypes[0])).isPresentAndNonNull();
+                                  if (parameterTypes.length != numOptionParams) return false;
```

Comment:
This looks familiar for another PR (https://github.com/apache/incubator-brooklyn/pull/639) - will assume that is going to be reviewed+merged first, and this PR rebased.